app/services/statement_creator.rb

```diff
@@ -0,0 +1,71 @@
+class StatementCreator
+
+  attr_accessor :order_detail_ids, :errors, :to_statement, :account_statements, :session_user, :current_facility
+
+  def initialize(params)
+    @order_detail_ids = params[:order_detail_ids]
+    @session_user = params[:session_user]
+    @current_facility = params[:current_facility]
+    @errors = []
+    @to_statement = {}
+  end
+
+  def create
+    OrderDetail.transaction do
+      set_order_details_to_statement
+      setup_statement_from_details
+      raise ActiveRecord::Rollback if errors.any?
+    end
+
+    errors.none?
+  end
+
+  def formatted_errors
+    errors.join("<br/>")
+  end
+
+  def send_statement_emails
+    if SettingsHelper.feature_on?(:send_statement_emails)
+      account_statements.each do |account, statement|
+        account.notify_users.each { |u| Notifier.delay.statement(user: u, facility: statement.facility, account: account, statement: statement) }
+      end
+    end
+  end
+
+  def account_list
+    account_statements.map { |a, _s| a.account_list_item }
+  end
+
+  def formatted_account_list
+    account_list.join("<br/>")
+  end
+
+  private
+
+  def set_order_details_to_statement
+    order_detail_ids.each do |order_detail_id|
+      od = nil
+      begin
+        od = OrderDetail.need_statement(current_facility).readonly(false).find(order_detail_id)
+        @to_statement[od.account] ||= []
```

Comment:
I think if we change the intializer to `@to_statement = Hash.new([])` so it defaults to empty array, I think we can get rid of this. Don't necessarily need to do this now. I would be comfortable holding off on further refactors until we get to DC's one-email-per-generation.